### DISCUSSION TITLE
computation of res.F

### DISCUSSION THREAD

**Discussion by sarapuccia1** [anyoptimization/pymoo#485] [Q&A]
Hi everybody, 
I have a question about how ref.F is computed for NSGA-II. The results that I have are considering only the last generation? Or does the code extrapolate the values through the optimization process? 

I have this issue since I tried to obtain the pareto also in a second way considering only the objective functions of the last generations and the solution is not the same with res.F.
Can you please explain me why ? 

Thanks and regards,
Sara

**Comment by blankjul** [comment]
The result is only the last generation of the algorithm. Since algorithms are designed to improve the solution set from generation to generation this is basically the best approximation that could be found in a run.

**Comment by blankjul** [comment]
What are you referring to when you say "second way"?
